pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -42,12 +42,16 @@ func (m *kubeGenericRuntimeManager) generateContainerConfig(container *api.Conta
 		return nil, err
 	}
 
-	_, containerName, cid := buildContainerName(pod, container)
+	name := buildContainerName(pod, container.Name)
+	// Create a randomly generated ID for the container in order to uniquely
+	// identify each container instance. Note that this ID is different from
+	// the container ID runtime returns *after* creation.
+	cid := makeUID()
```

Comment:
(pod_name,  pod_namespace) is actually not unique since a pod can be recreated in a short period of time. The randomly generated ID makes the possibility of having name collision lower, so we can probably ignore that. If we switch this to restart count, we'll have to add pod UID to the name to avoid collision.